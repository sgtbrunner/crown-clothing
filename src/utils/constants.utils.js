// INITIAL STATE
export const DEFAULT_INITIAL_INPUT_FIELD_STATE = {
  isValid: false,
  value: '',
  showError: false,
};

// FORM STRINGS
export const DISPLAY_NAME = 'displayName';
export const EMAIL = 'email';
export const PASSWORD = 'password';
export const CONFIRM_PASSWORD = 'confirmPassword';

// ERROR MESSAGES
export const DISPLAY_NAME_ERROR_MESSAGE =
  'Display name must have at least 3 characters';
export const EMAIL_ERROR_MESSAGE = 'Please enter a valid email address';
export const PASSWORD_ERROR_MESSAGE =
  'Your password must be at least 6 characters long';
export const CONFIRM_PASSWORD_ERROR_MESSAGE =
  'Password and confirm password do not match';
export const UNKNOWN_ERROR =
  'An unknown error occurred. Please try again later!';

// OTHER
export const EMPTY_CART_MESSAGE = 'Your cart is empty...';
export const BUTTON = 'button';
