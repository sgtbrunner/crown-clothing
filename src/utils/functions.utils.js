const EMAIL_REGEX_KEY = /^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i;

export const capitalizeFirstLetter = (string) =>
  string[0].toUpperCase() + string.slice(1);

export const isEmailValid = (email) =>
  EMAIL_REGEX_KEY.test(String(email).toLowerCase());
