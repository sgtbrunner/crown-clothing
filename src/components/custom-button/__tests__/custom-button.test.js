import React from 'react';
import { render, screen } from '@testing-library/react';

import { BUTTON } from '../../../utils/constants.utils';
import CustomButton from '../custom-button.component';

const props = {
  children: 'Custom Button',
};

const mountComponent = (newProps = {}) =>
  render(<CustomButton {...props} {...newProps} />);

describe('CustomButton component', () => {
  it('should render component snapshot', () => {
    const { container } = mountComponent();

    expect(container).toMatchSnapshot();
  });

  it('should render component label accordingly', () => {
    mountComponent();

    const button = screen.getByRole(BUTTON, { name: props.children });
    expect(button).toBeInTheDocument();
  });
});
