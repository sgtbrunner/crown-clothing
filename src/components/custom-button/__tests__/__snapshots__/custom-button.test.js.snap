// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton component should render component snapshot 1`] = `
<div>
  <button
    class="sc-bdnxRM JsWqM"
  >
    Custom Button
  </button>
</div>
`;
